_source/_posts/2019-10-21-illustrated-guide-to-oauth-and-oidc.md

```diff
@@ -0,0 +1,158 @@
+
+1. Pick your email provider
+1. Redirect to your email provider and login if needed
+1. Give "Terrible Pun of the Day" permission to access to your contacts
+1. Redirect back to "Terrible Pun of the Day"
+
+> In case you change your mind, applications that use OAuth to grant access also provide a way to revoke access. Should you decide later you no longer want your contacts shared, you can go to your email provider and remove "Terrible Pun of the Day" as an authorized application.
+
+### Let the OAuth Flow
+
+You've just stepped through what is commonly referred to as an OAuth _flow_. The OAuth flow in this example is made of visible steps to _grant consent_, as well as some invisible steps where the two services agree on a secure way of exchanging information. The previous "Terrible Pun of the Day" example uses the most common OAuth 2.0 flow, known as the "authorization code" flow.
+
+Before we dive into more details on what OAuth is doing, let's map some of the OAuth terminologies.
+
+<table>
+<tr>
+    <td>{% img blog/illustrated-guide-to-oauth-and-oidc/resource-owner.jpg alt:"Resource Owner" width:"200" %}</td>
+    <td markdown="span">**Resource Owner**: You! You are the owner of your identity, your data, and any actions that can be performed with your accounts.</td>
+</tr>
+<tr>
+    <td>{% img blog/illustrated-guide-to-oauth-and-oidc/client.jpg alt:"Client" width:"200" %}</td>
+    <td markdown="span">**Client**: The application (e.g. "Terrible Pun of the Day") that wants to access data or perform actions on behalf of the **Resource Owner**.</td>
+</tr>
+<tr>
+    <td>{% img blog/illustrated-guide-to-oauth-and-oidc/authorization-server.jpg alt:"Authorization Server" width:"200" %}</td>
+    <td markdown="span">**Authorization Server**: The application that knows the **Resource Owner**, where the **Resource Owner** already has an account.</td>
+</tr>
+<tr>
+    <td>{% img blog/illustrated-guide-to-oauth-and-oidc/resource-server.jpg alt:"Resource Server" width:"200" %}</td>
+    <td markdown="span">**Resource Server**: The Application Programming Interface (API) or service the **Client** wants to use on behalf of the **Resource Owner**.</td>
+</tr>
+<tr>
+    <td>{% img blog/illustrated-guide-to-oauth-and-oidc/redirect-uri.jpg alt:"Redirect URI" width:"200" %}</td>
+    <td markdown="span">**Redirect URI**: The URL the **Authorization Server** will redirect the **Resource Owner** back to after granting permission to the **Client**. This is sometimes referred to as the "Callback URL."</td>
+</tr>
+<tr>
+    <td>{% img blog/illustrated-guide-to-oauth-and-oidc/response-type.jpg alt:"Response/Grant Type" width:"200" %}</td>
+    <td markdown="span">**Response/Grant Type**: The type of authorization flow the **Client** wants to use. The most common type is `code`, where the **Client** will receive an **Authorization Code** and exchange it for an **Access Token**.</td>
```

Comment:
It may be too early / confusing to get into this here, but the "response_type" is what you expect to get back in this leg of the interaction.

`response_type=code` means you expect to get back an authorization code

in the second leg, `grant_type` is used to indicate the flow: `authorization_code`

Reply:
@dogeared Would this work? Is there a less confusing way of introducing this information? 

**Response/Grant Type**: The type of authorization flow the **Client** wants to use. The most common type is `code`, where the **Client** will request an **Authorization Code** (response type) and exchange it for an **Access Token** (grant type).